src/Hexagons.jl

```diff
@@ -291,17 +285,18 @@ end
 spiral(hex::Hexagon, n::Int) = spiral(n, hex)
 
 length(it::HexagonSpiralIterator) = it.n * (it.n + 1) * 3
-function start(it::HexagonSpiralIterator)
+function _start(it::HexagonSpiralIterator)
     first_ring = ring(it.hex, 1)
     HexagonSpiralIteratorState(1, first_ring, start(first_ring)...)
 end
-done(it::HexagonSpiralIterator, state::HexagonSpiralIteratorState) = state.hexring_i > it.n
 
 # The state of a HexagonSpiralIterator consists of
 # 1. an Int, the index of the current ring
 # 2. a HexagonRingIterator and its state to keep track of the current position
 #    in the ring.
-function next(it::HexagonSpiralIterator, state::HexagonSpiralIteratorState)
+function Base.iterate(it::HexagonSpiralIterator,
+                      state::HexagonSpiralIteratorState=_start(it))
```

Comment:
are their pros and cons to calling `_start` in the function signature, versus creating a second `iterate` method with a single input argument as i did in #11 ?  the latter is easier to read to me, but i'm not familiar enough with julia internals to know if there's a performance difference.